ci_framework/roles/operator_build/tasks/build.yml

```diff
@@ -0,0 +1,147 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Set default api path for {{ operator.name }}
+  ansible.builtin.set_fact:
+    operator_api_path: "github.com/{{ cifmw_operator_build_org }}/{{ operator.name }}/api"
+
+- name: Update the go.mod file in meta operator for provided PR_SHA
+  ansible.builtin.shell: |
+    go mod edit -replace {{ operator_api_path }}=github.com/{{ operator.pr_owner }}/api@{{ operator.pr_sha }}
+    go mod tidy
+  args:
+    chdir: "{{ cifmw_operator_build_meta_src }}"
+  when:
+    - cifmw_operator_build_meta_build
+    - operator.name != cifmw_operator_build_meta_name
+    - operator.pr_owner is defined
+    - operator.pr_sha is defined
+
+- name: Get the {{ operator.name }} latest commit when no PR is provided
+  ansible.builtin.command:
+    cmd: git show-ref --head --hash head
+    chdir: "{{ operator.src }}"
+  register: git_head_out
+  when:
+    - operator.pr_sha is not defined
+
+- name: Set pr_sha to be used as image tag
+  ansible.builtin.set_fact:
+    pr_sha: "{{ operator.pr_sha | default(git_head_out.stdout | trim) }}"
+
+- name: Update the go.mod file using latest commit if no PR is provided
+  ansible.builtin.shell: |
+    go mod edit -replace {{ operator_api_path }}={{ operator_api_path }}@{{ pr_sha }}
+    go mod tidy
+  args:
+    chdir: "{{ cifmw_operator_build_meta_src }}"
+  when:
+    - cifmw_operator_build_meta_build
+    - operator.name != cifmw_operator_build_meta_name
+    - pr_sha is defined
+    - operator.pr_owner is not defined
+
+- name: Get golang container image
+  containers.podman.podman_image:
+    name: docker.io/library/golang:1.19
+
+- name: Set operator image names and tags
+  ansible.builtin.set_fact:
+    operator_img_tag: "{{ pr_sha }}"
+    operator_registry_prefix: "{{ cifmw_operator_build_push_registry }}/{{ cifmw_operator_build_push_org }}/{{ operator.name }}"
+    cacheable: true
+
+- name: Create artifact directory
+  ansible.builtin.file:
+    path: /tmp/artifacts
+    state: directory
+
+- name: Call manifests
+  ci_make:
+    chdir: "{{ operator.src }}"
+    output_dir: /tmp/artifacts
+    target: manifests
+  tags:
+    - build
```

Comment:
we may not need build that if we add ci_make dry-run functionality.
we can continue with `push` tag, since we may not need to always push to some registry, so we would skip it.

Reply:
ah - I removed the "push" tag - will re-add it, but maybe as a parameter ?

Reply:
added a parameter instead, updated molecule.